pkg/specgen/namespaces.go

```diff
@@ -514,6 +515,9 @@ func SetupUserNS(idmappings *storageTypes.IDMappingOptions, userns Namespace, g
 		if err != nil {
 			return user, err
 		}
+		if opts.MaxSize != nil && !rootless.IsRootless() {
+			return user, fmt.Errorf("cannot set max size for user namespace when not running rootless")
+		}
```

Comment:
Why not? Could we not just limit the rootful size in the same way as done for rootless?

Reply:
keep-id with root is a no-op.  It doesn't currently create a user namespace